src/algebraic_geometry/EllipticCurve/weierstrass.lean

```diff
@@ -0,0 +1,287 @@
+  a₆ := ↑u⁻¹ ^ 6 * (C.a₆ + r * C.a₄ + r ^ 2 * C.a₂ + r ^ 3 - t * C.a₃ - t ^ 2 - r * t * C.a₁) }
+
+@[simp] lemma variable_change_b₂ : (C.variable_change u r s t).b₂ = ↑u⁻¹ ^ 2 * (C.b₂ + 12 * r) :=
+by { simp only [b₂, variable_change_a₁, variable_change_a₂], ring1 }
+
+@[simp] lemma variable_change_b₄ :
+  (C.variable_change u r s t).b₄ = ↑u⁻¹ ^ 4 * (C.b₄ + r * C.b₂ + 6 * r ^ 2) :=
+by { simp only [b₂, b₄, variable_change_a₁, variable_change_a₃, variable_change_a₄], ring1 }
+
+@[simp] lemma variable_change_b₆ :
+  (C.variable_change u r s t).b₆ = ↑u⁻¹ ^ 6 * (C.b₆ + 2 * r * C.b₄ + r ^ 2 * C.b₂ + 4 * r ^ 3) :=
+by { simp only [b₂, b₄, b₆, variable_change_a₃, variable_change_a₆], ring1 }
+
+@[simp] lemma variable_change_b₈ :
+  (C.variable_change u r s t).b₈
+    = ↑u⁻¹ ^ 8 * (C.b₈ + 3 * r * C.b₆ + 3 * r ^ 2 * C.b₄ + r ^ 3 * C.b₂ + 3 * r ^ 4) :=
+by { simp only [b₂, b₄, b₆, b₈, variable_change_a₁, variable_change_a₂, variable_change_a₃,
+                variable_change_a₄, variable_change_a₆], ring1 }
+
+@[simp] lemma variable_change_c₄ : (C.variable_change u r s t).c₄ = ↑u⁻¹ ^ 4 * C.c₄ :=
+by { simp only [c₄, variable_change_b₂, variable_change_b₄], ring1 }
+
+@[simp] lemma variable_change_c₆ : (C.variable_change u r s t).c₆ = ↑u⁻¹ ^ 6 * C.c₆ :=
+by { simp only [c₆, variable_change_b₂, variable_change_b₄, variable_change_b₆], ring1 }
+
+@[simp] lemma variable_change_Δ : (C.variable_change u r s t).Δ = ↑u⁻¹ ^ 12 * C.Δ :=
+by { dsimp, ring1 }
+
+end variable_change
+
+section base_change
+
+/-! ### Base changes -/
+
+variables (A : Type v) [comm_ring A] [algebra R A]
+
+/-- The Weierstrass curve over `R` base changed to `A`. -/
+@[simps] def base_change : weierstrass_curve A :=
+⟨algebra_map R A C.a₁, algebra_map R A C.a₂, algebra_map R A C.a₃, algebra_map R A C.a₄,
+algebra_map R A C.a₆⟩
+
+@[simp] lemma base_change_b₂ : (C.base_change A).b₂ = algebra_map R A C.b₂ :=
+by { simp only [b₂, base_change_a₁, base_change_a₂], map_simp }
+
+@[simp] lemma base_change_b₄ : (C.base_change A).b₄ = algebra_map R A C.b₄ :=
+by { simp only [b₄, base_change_a₁, base_change_a₃, base_change_a₄], map_simp }
+
+@[simp] lemma base_change_b₆ : (C.base_change A).b₆ = algebra_map R A C.b₆ :=
+by { simp only [b₆, base_change_a₃, base_change_a₆], map_simp }
+
+@[simp] lemma base_change_b₈ : (C.base_change A).b₈ = algebra_map R A C.b₈ :=
+by { simp only [b₈, base_change_a₁, base_change_a₂, base_change_a₃, base_change_a₄, base_change_a₆],
+     map_simp }
+
+@[simp] lemma base_change_c₄ : (C.base_change A).c₄ = algebra_map R A C.c₄ :=
+by { simp only [c₄, base_change_b₂, base_change_b₄], map_simp }
+
+@[simp] lemma base_change_c₆ : (C.base_change A).c₆ = algebra_map R A C.c₆ :=
+by { simp only [c₆, base_change_b₂, base_change_b₄, base_change_b₆], map_simp }
+
+@[simp] lemma base_change_Δ : (C.base_change A).Δ = algebra_map R A C.Δ :=
+by { simp only [Δ, base_change_b₂, base_change_b₄, base_change_b₆, base_change_b₈], map_simp }
```

Comment:
Maybe `simp` takes too long to simplify the LHS, so the linter fails. The fix is probably `@[nolint simp_nf]`.